### PR TITLE
Migrate std::bind calls to lambda expressions

### PR DESCRIPTION
**Pull request by FelipeGdM**
This PR is a follow up to the changes made in the documentation https://github.com/ros2/ros2_documentation/pull/4041

The migration of std::bind calls made in the docs is reflected here

### CODE REVIEW

**Review by ahcorde — APPROVED** [review]

File: turtle_tf2_cpp/src/turtle_tf2_broadcaster.cpp [hunk 1]
```diff
@@ -42,42 +42,40 @@ class FramePublisher : public rclcpp::Node
     stream << "/" << turtlename_.c_str() << "/pose";
     std::string topic_name = stream.str();
 
-    subscription_ = this->create_subscription<turtlesim::msg::Pose>(
-      topic_name, 10,
-      std::bind(&FramePublisher::handle_turtle_pose, this, std::placeholders::_1));
-  }
+    auto handle_turtle_pose = [this](const std::shared_ptr<turtlesim::msg::Pose> msg){
```
Comment (clalancette) [inline turtle_tf2_cpp/src/turtle_tf2_broadcaster.cpp]:
```suggestion
    auto handle_turtle_pose = [this](const std::shared_ptr<turtlesim::msg::Pose> msg) {
```

File: turtle_tf2_cpp/src/turtle_tf2_broadcaster.cpp [hunk 2]
```diff
@@ -42,42 +42,40 @@ class FramePublisher : public rclcpp::Node
...
-    t.transform.rotation.x = q.x();
-    t.transform.rotation.y = q.y();
-    t.transform.rotation.z = q.z();
-    t.transform.rotation.w = q.w();
-
-    // Send the transformation
-    tf_broadcaster_->sendTransform(t);
+        // Send the transformation
+        tf_broadcaster_->sendTransform(t);
+    };
```
Comment (clalancette) [inline turtle_tf2_cpp/src/turtle_tf2_broadcaster.cpp]:
```suggestion
      };
```

**Review by clalancette — CHANGES_REQUESTED** [review]
Two changes to pass linters, then this is probably good to go.

File: turtle_tf2_cpp/src/turtle_tf2_listener.cpp [hunk 3]
```diff
@@ -53,7 +53,7 @@ class FrameListener : public rclcpp::Node
 
     // Call on_timer function every second
     timer_ = this->create_wall_timer(
-      1s, std::bind(&FrameListener::on_timer, this));
+      1s, [this](){return this->on_timer();});
```
Comment (clalancette) [inline turtle_tf2_cpp/src/turtle_tf2_listener.cpp]:
```suggestion
      1s, [this]() {return this->on_timer();});
```

**Review by clalancette — CHANGES_REQUESTED** [review]
One more linter error (not sure why this didn't show up in the earlier run).

**Review by ahcorde — CHANGES_REQUESTED** [review]
```bash
2024-01-08T15:02:27.9197222Z 7: --- src/turtle_tf2_listener.cpp
2024-01-08T15:02:27.9197566Z 7: +++ src/turtle_tf2_listener.cpp.uncrustify
2024-01-08T15:02:27.9197935Z 7: @@ -56 +56 @@
2024-01-08T15:02:27.9198266Z 7: -      1s, [this](){return this->on_timer();});
2024-01-08T15:02:27.9198687Z 7: +      1s, [this]() {return this->on_timer();});
```

**Comment by FelipeGdM** [comment]
I executed `ament_uncrustify src/` locally to be sure all format errors are solved >.<

**Pull request merged**